test/module.lean

```diff
@@ -0,0 +1,306 @@
+#guard_msgs in
+example : x ≠ y := by module
+
+end Nat
+
+/-! ### `ℤ` (most tests copied from the `abel` tactic) -/
+
+variable [AddCommGroup V]
+
+example : (x + y) - ((y + x) + x) = -x := by module
+example : x - 0 = x := by module
+example : (3 : ℤ) • x = x + (2 : ℤ) • x := by module
+example : x - 2 • y = x - 2 • y := by module
+example : (x + y) - ((y + x) + x) = -x := by module
+example : x + y + (z + w - x) = y + z + w := by module
+example : x + y + z + (z - x - x) = (-1) • x + y + 2 • z := by module
+example : -x + x = 0 := by module
+example : x - (0 - 0) = x := by module
+example : x + (y - x) = y  := by module
+example : -y + (z - x) = z - y - x := by module
+
+example : x + y = y + x ∧ (↑((1:ℕ) + 1) : ℚ) = 2 := by
+  constructor
+  module -- do not focus this tactic: the double goal is the point of the test
+  guard_target =ₐ (↑((1:ℕ) + 1) : ℚ) = 2
+  norm_cast
+
+-- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Interaction.20of.20abel.20with.20casting/near/319895001
+example : True := by
+  have : ∀ (p q r s : V), s + p - q = s - r - (q - r - p) := by
+    intro p q r s
+    module
+  trivial
+
+example : True := by
+  have : ∀ (p q r s : V), s + p - q = s - r - (q - r - p) := by
+    intro p q r s
+    match_scalars
+    · decide
+    · decide
+    · decide
+    · decide
+  trivial
+
+-- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Interaction.20of.20abel.20with.20casting/near/319897374
+example : y = x + z - (x - y + z) := by
+  have : True := trivial
+  module
+
+example : y = x + z - (x - y + z) := by
+  have : True := trivial
+  match_scalars <;> decide
+
+/--
+error: unsolved goals
+V : Type u_1
+K : Type u_2
+t u v w x y z : V
+a b c d e f μ ν ρ : K
+inst✝ : AddCommGroup V
+⊢ -1 + 1 = 0
+-/
+#guard_msgs in
+example : -x + x = 0 := by
+  match_scalars
+
+/-! ### Commutative ring -/
+
+section CommRing
+variable [CommRing K] [Module K V]
+
+example : a • x + b • x = (a + b) • x := by module
+example : a • x - b • x = (a - b) • x := by module
+example : a • x - b • y = a • x + (-b) • y := by module
+example : 2 • a • x = a • 2 • x := by module
+example : a • x - b • y = a • x + (-b) • y := by module
+example : (μ - ν) • a • x = (a • μ • x + b • ν • y) - ν • (a • x + b • y) := by module
+example : (μ - ν) • b • y = μ • (a • x + b • y) - (a • μ • x + b • ν • y) := by module
+
+-- from https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/smul.20diamond/near/457163013
+example : (4 : ℤ) • v = (4 : K) • v := by module
+example : (4 : ℕ) • v = (4 : K) • v := by module
+
+-- from https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/linear_combination.20for.20groups/near/437042918
+example : (1 + a ^ 2) • (v + w) - a • (a • v - w) = v + (1 + a + a ^ 2) • w := by module
+
+example (h : a = b) : a • x = b • x := by
+  match_scalars
+  linear_combination h
+
+example (h : a = b) : a • x = b • x := by
+  -- `linear_combination h • x`
```

Comment:
Could you add another sentence explaining this comment?

Reply:
I did this here (just once, not in all the similar places -- let me know if you would like the change throughout).